lib/generators/scenic/view/view_generator.rb

```diff
@@ -7,6 +7,20 @@ class ViewGenerator < Rails::Generators::NamedBase
       include Rails::Generators::Migration
       source_root File.expand_path("../templates", __FILE__)
 
+      def split_name_and_arguments
+        args = name.split(":")
+        @name_with_arguments = name
+        self.name = args.shift
+        assign_names!(name)
+
+        args.each do |arg|
+          setter = "#{arg}="
+          if respond_to?(setter)
+            send(setter, true)
+          end
+        end
```

Comment:
This is entirely duplicated in the generator above. If we're going to handle arguments this way, we should extract a parser object for this.